packages/devtools/devtools/src/components/MasterDetailTable.tsx

```diff
@@ -22,21 +23,35 @@ export const MasterDetailTable = <T extends {}>({
   pinToBottom,
   widths = ['w-1/2', 'w-1/2'],
 }: MasterTableProps<T>) => {
-  const [selected, setSelected] = useState<T[]>();
+  const [selected, setSelected] = useState<T>();
 
   return (
     <div className='flex grow overflow-hidden divide-x'>
-      <div className={mx('flex overflow-hidden', widths[0])}>
-        <Table<T>
-          columns={columns}
-          data={data}
-          selected={selected}
-          onSelectedChange={setSelected}
-          pinToBottom={pinToBottom}
-        />
-      </div>
-
-      <div className={mx('flex overflow-auto', widths[1])}>{selected && <JsonView data={selected?.[0]} />}</div>
+      {pinToBottom ? (
+        <AnchoredOverflow.Root classNames={widths[0]}>
+          <Table<T>
+            columns={columns}
+            data={data}
+            rowsSelectable
+            currentDatum={selected}
+            onDatumClick={setSelected}
+            fullWidth
+          />
+          <AnchoredOverflow.Anchor />
+        </AnchoredOverflow.Root>
+      ) : (
+        <div className={mx('overflow-auto', widths[0])}>
+          <Table<T>
```

Comment:
I'm fine with this, but say this was more complicated (e.g., many more params) is there a zero-runtime-overhead way to "factor out" the inner Table def so that it isn't repeated?

Reply:
Now that you mention it, yes, absolutely. Will adjust.